DTML.EduBot/Dialogs/RootDialog.cs

```diff
@@ -1,31 +1,103 @@
-﻿namespace DTML.EduBot.Dialogs
+﻿using Autofac;
+using Microsoft.Bot.Builder.Internals.Fibers;
+
+namespace DTML.EduBot.Dialogs
 {
     using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Linq;
     using System.Threading.Tasks;
     using Microsoft.Bot.Builder.Dialogs;
-    using Microsoft.Bot.Builder.Luis;
-    using Microsoft.Bot.Builder.Luis.Models;
-    using DTML.EduBot.Common;
-    using DTML.EduBot.Qna;
+    using Microsoft.Bot.Connector;
 
-    [LuisModel("", "")]
-    [QnaModel(hostUri: "", subscriptionKey: "", modelId: "")]
     [Serializable]
-    public partial class RootDialog : QnaLuisDialog<object>
+    public class RootDialog : IDialog<string>
     {
-        [LuisIntent("")]
-        [LuisIntent("None")]
-        public async Task HandleUnrecognizedIntent(IDialogContext context, LuisResult result)
+        private const string ChatWithBot = "Chat With Bot";
+        private const string StartTheLessonPlan = "Start The Lesson Plan";
+
+        private static readonly IEnumerable<string> Choices = new ReadOnlyCollection<string>
+            (new List<String> {
+                ChatWithBot,
+                StartTheLessonPlan});
+
+        private readonly LessonPlanDialog lessonPlanDialog;
+
+        public RootDialog()
+        {
+            using (var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
+            {
+                lessonPlanDialog = WebApiApplication.FindContainer().Resolve<LessonPlanDialog>();
```

Comment:
Why did we need a container to resolve the dependency for LessonPlanDialog here? Can you inject this dependency from constructor instead?